src/experimental/constructs/iam/roles/cross-account-role.test.ts

```diff
@@ -0,0 +1,62 @@
+import { Template } from "aws-cdk-lib/assertions";
+import { ServicePrincipal } from "aws-cdk-lib/aws-iam";
+import { GuRole } from "../../../../constructs/iam";
+import { simpleGuStackForTesting } from "../../../../utils/test";
+import { GuCrossAccountRoleExperimental } from "./cross-account-role";
+
+describe("The GuCrossAccountRoleExperimental construct", () => {
+  it("can create a cross account role", () => {
+    const stack = simpleGuStackForTesting();
+    new GuCrossAccountRoleExperimental(stack, "testCrossAccountRole", {
+      nameOfRoleWhichCanAssumeThisRole: "nameOfRoleInOtherAccountWhichCanAssumeThis-STAGE",
+      roleName: "crossAccountRole",
+      accountId: "1234",
+    });
+    expect(Template.fromStack(stack).toJSON()).toMatchSnapshot();
+  });
+
+  it("can create a cross account role that can be assumed by a service in another account", () => {
+    const stackThatCreatesTheRole = simpleGuStackForTesting();
+    new GuCrossAccountRoleExperimental(stackThatCreatesTheRole, "testCrossAccountRole", {
+      nameOfRoleWhichCanAssumeThisRole: "nameOfRoleInOtherAccountWhichCanAssumeThisNewlyCreatedOne-CODE",
+      roleName: "crossAccountRole",
+      accountId: "1234",
+    });
+
+    Template.fromStack(stackThatCreatesTheRole).hasResourceProperties("AWS::IAM::Role", {
+      RoleName: "crossAccountRole",
+      AssumeRolePolicyDocument: {
+        Statement: [
+          {
+            Action: "sts:AssumeRole",
+            Effect: "Allow",
+            Principal: {
+              AWS: "arn:aws:iam::1234:role/nameOfRoleInOtherAccountWhichCanAssumeThisNewlyCreatedOne-CODE",
+            },
+          },
+        ],
+      },
+    });
+
+    const stackThatAssumesTheRole = simpleGuStackForTesting();
+    new GuRole(stackThatAssumesTheRole, "idForRole", {
+      assumedBy: new ServicePrincipal("ec2.amazonaws.com"),
+      roleName: "nameOfRoleInOtherAccountWhichCanAssumeThisNewlyCreatedOne-CODE",
+    });
+
```

Comment:
I think we may need to add an instance of `GuAssumeRolePolicy` (via [`addToPolicy`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_iam.Role.html#addwbrtowbrpolicystatement)) to complete this example?

Reply:
And then I think the important thing for our assertion to check is that it produces an IAM role that looks something like [this](https://github.com/guardian/mobile-notifications-content/blob/f6272f6cd4e48bfaa43a740973f7ebf535697da8/cfn.yaml#L65-L70)?